### PR TITLE
[backend] Add a ProtectUser in place of a User in the execution user …

### PR DESCRIPTION
**Pull request by guillaumejparis**
…context and add pgpkey in the user execution context (#320)

### CODE REVIEW

**Comment by guillaumejparis** [comment]
https://github.com/OpenEx-Platform/injectors/pull/28

File: openex-api/src/main/java/io/openex/helper/InjectHelper.java [hunk 1]
```diff
@@ -30,92 +32,93 @@
...
+              .map(user -> Tuples.of(user, audience.getName())));
     }
+    throw new UnsupportedOperationException("Unsupported type of Injection");
+  }
 
-    private List<ExecutionContext> usersFromInjection(Injection injection) {
-        return getUsersFromInjection(injection)
-                .collect(groupingBy(ExecutionContext::getUser)).entrySet().stream()
-                .map(entry -> new ExecutionContext(openExConfig, entry.getKey(), injection,
-                        entry.getValue().stream().flatMap(ua -> ua.getAudiences().stream()).toList()))
```
Comment (guillaumejparis) [inline openex-api/src/main/java/io/openex/helper/InjectHelper.java:83]: We had a problem here with the ExecutionContext::getUser that was a ProtectUser and when we gave it to the ExecutionContext we had a type problem as ExecutionContext  want a User
So we remove the ExectutionContext creation in getUsersFromInjection and just pass a tuple of user & audience

File: openex-api/src/main/java/io/openex/injects/email/service/EmailPgp.java [hunk 2]
```diff
@@ -23,57 +23,57 @@
 @Component
 public class EmailPgp {
 
-    public PGPPublicKey getUserPgpKey(User user) throws IOException {
```
Comment (guillaumejparis) [inline openex-api/src/main/java/io/openex/injects/email/service/EmailPgp.java:26]: Bingo ! It seems that we had a problem on the sending an email with a pgp key too

File: openex-injectors
Comment (guillaumejparis) [inline openex-injectors:1]: https://github.com/OpenEx-Platform/injectors/pull/28

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**